src/services/entry.rs

```diff
@@ -10,7 +10,20 @@ pub fn get_all_entries(conn: DBPool) -> Vec<Entry> {
 }
 
 pub fn insert_entry(entry: NewEntry, user_id: Uuid, conn: DBPool) -> Result<Entry, &'static str> {
-    add_new(entry, user_id, &conn).ok_or("cannot insert entry")
+    let photo = ImageService::create_from_base(entry.photo, &conn);
+    match photo {
+        Err(s) => Err(s),
+        Ok(p) => {
+            let new_entry = NewEntry {
+                photo: ImageService::gen_link(p),
+                name: entry.name,
+                voltage: entry.voltage,
+                price: entry.price,
+                volume: entry.volume 
+            };
+            add_new(new_entry, user_id, &conn).ok_or("cannot insert entry")
+        }
+    }
```

Comment:
```suggestion
    ImageService::create_from_base(entry.photo, &conn)
    .map(|file| ImageService::gen_link(file)  )
    .map(|link| 
        NewEntry {
            name: entry.name,
            voltage: entry.voltage,
            price: entry.price,
            volume: entry.volume,
            photo: link
    })
    .and_then( |entry| add_new(entry, user_id, &conn) )
```

Reply:
It would also require a change of `add_new` return type from `Option` to `Result`